tools/test.py

```diff
@@ -44,13 +49,49 @@ def run_tests(tests, bin_path, test_args=[], algorithms=[]):
                 print(" ".join(test_cmd))
                 subprocess.run(test_cmd, check=True, shell=True, env=my_env)
 
+                if coverage:
+                    generate_report(test_name, my_env)
+
+
+def generate_report(test, env):
+    print(f"Generating coverage report for {test}.")
+    subprocess.run(["mkdir", "-p", f"coverage/{test}/html"], env=env)
+    subprocess.run([
+        "llvm-profdata",
```

Comment:
We should check somewhere that these commands exist before using them.

Reply:
Agree. Will do.